tests/test_graph.py

```diff
@@ -12,36 +15,151 @@
+        # pytest output will CUT OFF things, make sure we see the full list!
+        # PLEASE NOTE THIS PROPAGATES THE ERROR AFTERWARDS, we're not
+        # actually catching something here, just adding output.
+        print("test_valid_recipe_order_and_bootstrap FAILED "
+              "with these specific parameters: {}".format(
+                  (ctx, names, bootstrap))
+             )
+        raise e
 
 
 @pytest.mark.parametrize('names,bootstrap', invalid_combinations)
 def test_invalid_recipe_order_and_bootstrap(names, bootstrap):
     with pytest.raises(BuildInterruptingException) as e_info:
         get_recipe_order_and_bootstrap(ctx, names, bootstrap)
-    assert e_info.value.message == (
-        "Didn't find any valid dependency graphs. "
-        "This means that some of your requirements pull in conflicting dependencies."
+    assert "conflict" in e_info.value.message.lower()
+
+
+@pytest.mark.parametrize('names,bootstrap', valid_combinations)
+def test_valid_obvious_conflict_checker(names, bootstrap):
+    try:
+        # Note: obvious_conflict_checker is stricter on input
+        # (needs fix_deplist) than get_recipe_order_and_bootstrap!
+        obvious_conflict_checker(ctx, fix_deplist(names))
+    except Exception as e:
+        # pytest output will CUT OFF things, make sure we see the full list!
+        # PLEASE NOTE THIS PROPAGATES THE ERROR AFTERWARDS, we're not
+        # actually catching something here, just adding output.
+        print("test_valid_obvious_conflict_checker FAILED "
+              "with these specific parameters: {}".format(
+                  (ctx, names, bootstrap))
+             )
+        raise e
+
+
+@pytest.mark.parametrize('names,bootstrap',
+                         invalid_combinations_simple  # see above for why this
+                        )                             # is a separate list
+def test_invalid_obvious_conflict_checker(names, bootstrap):
+    # Note: obvious_conflict_checker is stricter on input
+    # (needs fix_deplist) than get_recipe_order_and_bootstrap!
+    with pytest.raises(BuildInterruptingException) as e_info:
+        obvious_conflict_checker(ctx, fix_deplist(names))
+    assert "conflict" in e_info.value.message.lower()
+
+
+def test_misc_obvious_conflict_checker():
+    # Check that the assert about wrong input data is hit:
+    with pytest.raises(AssertionError) as e_info:
+        obvious_conflict_checker(
+            ctx,
+            ["this_is_invalid"]
+            # (invalid because it isn't properly nested as tuple)
+        )
+
+    # Test that non-recipe dependencies work in overall:
+    obvious_conflict_checker(
+        ctx, fix_deplist(["python3", "notarecipelibrary"])
     )
 
+    # Test that a conflict with a non-recipe dependency works:
+    # This is currently not used, so we need a custom test recipe:
+    # To get that, we simply modify one!
+    r = Recipe.get_recipe("python3", ctx)
+    old_conflicts = list(r.conflicts)
+    try:
```

Comment:
why, why is that?! why do we try in unit test like that? :sob: 
if the test fail it fails, right don't try/except in test please please :pray:

Reply:
@AndreMiras I'm letting it fail there is no `except` there :smile: it's just to get the recipe back into its old state no matter the test outcome

> don't try/except in test

i didn't :open_mouth:

Reply:
ahah OK I'll try to buy that then :smile: 
I'll not ask you to write a context manager just for 2 tests  :wink: 
Thanks!